oauth2/client_credentials_provider_test.go

```diff
@@ -34,12 +34,14 @@ func TestNewClientCredentialsProviderFromKeyFile(t *testing.T) {
 	ClientSecret := "CLIENT_SECRET"
 	ClientEmail := "CLIENT_EMAIL"
 	IssuerURL := "ISSUER_URL"
+	Scope := "SCOPE"
 	keyFile := &KeyFile{
 		Type:         oauthType,
 		ClientID:     clientID,
 		ClientSecret: ClientSecret,
 		ClientEmail:  ClientEmail,
 		IssuerURL:    IssuerURL,
+		Scope:        Scope,
```

Comment:
Could you help verify that the scope has been added to the `AdditionalScopes` successfully?

Reply:
I've pushed a change to the file client_credentials_flow_test.go that should verify the logic of adding the value of scope from keyfile to the `AdditionalScopes` change.

Reply:
I've reworked the code a bit so it now in the actual Token request, that way any usage of it has the logic.
My earlier implementation would have caused only the ones create earlier it would only happen for those created by the method NewDefaultClientCredentialsFlow and not newClientCredentialsFlow which is used in the provider test file.

A change to the client_credentials_flow_test.go was added that should verify, I however cannot run the test locally so currently only when it runs in the CI pipelines I can verify.

Reply:
Good work! The test works fine. Only some lint issues.